async-nats/src/lib.rs

```diff
@@ -604,13 +613,20 @@ impl ConnectionHandler {
             match process.await {
                 ExitReason::Disconnected(err) => {
                     debug!(?err, "disconnected");
-
                     if self.handle_disconnect().await.is_err() {
                         break;
                     };
                     debug!("reconnected");
                 }
                 ExitReason::Closed => break,
+                ExitReason::ReconnectRequested => {
+                    debug!("reconnect requested");
+                    // Should be ok to ingore error, as that means we are not in connected state.
+                    self.connection.stream.shutdown().await.ok();
```

Comment:
Doesn't matter with this PR in particular since the problem was always there but this should probably go inside `handle_disconnect`.